src/ec/packages/ec-component-link/link.html.twig

```diff
@@ -40,16 +40,19 @@
   size: ''
 }%}
 
-{% if icon is defined %}
+{% if icon.name is defined and icon.name is not empty %}
   {% set _icon = _icon|merge(icon) %}
+{% elseif icon is defined and icon is not empty and icon.name is not defined %}
```

Comment:
Just a thought: isn't it easier to check first of there is something in `icon`, then check if it is an array?

Reply:
The fact that there's no direct way of distinguishing an object from an array of objects (iterable works with both) is the reason of this weird logic here, i have to say that i've been trying diffferent approaches but probably not the easiest one, which would be checking for `icon[1] is not empty` to identify the array use case.

Reply:
indeed, `is iterable` isn't smart enough in this case. There seems to be different ways, like [this](https://stackoverflow.com/questions/13876023/check-if-variable-is-string-or-array-in-twig/29642662#29642662), but that's not really simpler (and I'm not sure we could do it in our case)